pkg/backend/identityapi/test/register.go

```diff
@@ -0,0 +1,89 @@
+package test
+
+import (
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
+	"github.com/newrelic/infrastructure-agent/pkg/backend/inventoryapi"
+	"github.com/newrelic/infrastructure-agent/pkg/backend/state"
+	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+)
+
+type EmptyRegisterClient struct{}
+
+func (icc *EmptyRegisterClient) RegisterEntitiesRemoveMe(agentEntityID entity.ID, entities []identityapi.RegisterEntity) (r []identityapi.RegisterEntityResponse, retryAfter time.Duration, err error) {
+	return
+}
+
+func (icc *EmptyRegisterClient) RegisterBatchEntities(agentEntityID entity.ID, entities []protocol.Entity) (r []identityapi.RegisterEntityResponse, retryAfter time.Duration, err error) {
+	return
+}
+
+func (icc *EmptyRegisterClient) RegisterEntity(agentEntityID entity.ID, entity protocol.Entity) (resp identityapi.RegisterEntityResponse, err error) {
+	return
+}
+
+type IncrementalRegister struct {
+	state state.Register
+}
+
+func NewIncrementalRegister() identityapi.RegisterClient {
+	return &IncrementalRegister{state: state.RegisterHealthy}
+}
+
+func NewRetryAfterRegister() identityapi.RegisterClient {
+	return &IncrementalRegister{state: state.RegisterRetryAfter}
+}
+
+func NewRetryBackoffRegister() identityapi.RegisterClient {
+	return &IncrementalRegister{state: state.RegisterRetryBackoff}
+}
+
+func (r *IncrementalRegister) RegisterBatchEntities(agentEntityID entity.ID, entities []protocol.Entity) (batchResponse []identityapi.RegisterEntityResponse, retryAfter time.Duration, err error) {
+	if r.state == state.RegisterRetryAfter {
+		retryAfter = 1 * time.Second
+		err = inventoryapi.NewIngestError("ingest service rejected the register step", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError), "")
+		return
+	} else if r.state == state.RegisterRetryBackoff {
+		err = inventoryapi.NewIngestError("ingest service rejected the register step", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError), "")
+		return
+	}
+
+	var i entity.ID
```

Comment:
Not sure that I can fully understand this `i` counter :/. Can you tell me please how it works here?

Reply:
got it, it is just a stub